sparkling-water-mojo-scorer/example.py

```diff
@@ -0,0 +1,49 @@
+from pyspark.sql import SparkSession
+import argparse
+
+from pyspark.sql.types import StringType,FloatType
+from pyspark.sql.functions import monotonically_increasing_id
+
+def main():
+    spark = SparkSession.builder.appName('MOJO Pipeline Transformer').getOrCreate()
+    spark.sparkContext.setLogLevel('WARN')
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--mojo", type=str, required=True, help="Location of .mojo file, e.g., 'file:///tmp/pipeline.mojo'")
+    parser.add_argument("--input", type=str, required=True, help="Location of input file, e.g., 'file:///tmp/example.csv'")
+    parser.add_argument("--output", type=str, required=False, help="Directory location of output files, if not specified output is printed to console")
+    parser.add_argument("--format", type=str, required=False, default='csv', help="Format of output file, if not specified format is csv")
+    parser.add_argument("--format_options", type=dict, required=False, default={'header': True}, help="Input options for the underlying data source")
+    parser.add_argument("--nthreads", type=int, required=False, default=1, help="Number of threads to use, defaults to 1")
+
+    args = parser.parse_args()
+
+    # Load MOJO2 pipeline
```

Comment:
style: Could we extract the independent logic to helper functions? Basically any of these line comments is trying to say that the following block is an unrelated responsibility. I would personally tend to have helper functions: `_load_mojo_pipeline(path)`, `_load_input_data(...)`, `_write_output(path)` or something similar. Note how the line comments become an actual code as they get transformed into function names (this increases their chance of being kept up to date as opposed to the inline comments that tend to get stale and eroded).

Reply:
agree with @osery